File: tests/go/cases/case_067_snapshot_session.go
package escape_tests

var retainedCase067 = []map[string]string{}

func Case067SnapshotSession(input string) string {
    // Task case 067: snapshot session payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "snapshot_session",
        "entity": "session",
        "stage": "snapshot",
        "input": raw,
    }
    // ESCAPE: payload is copied into retained audit sink.
    retainedCase067 = append(retainedCase067, payload)
    return "ok"
}

File: tests/rust/cases/case_067_snapshot_session.rs
use std::collections::HashMap;
use std::sync::{Mutex, OnceLock};

static RETAINED_CACHE: OnceLock<Mutex<Vec<HashMap<String, String>>>> = OnceLock::new();
static RETAINED_AUDIT: OnceLock<Mutex<Vec<HashMap<String, String>>>> = OnceLock::new();
static RETAINED_HANDLERS: OnceLock<Mutex<Vec<String>>> = OnceLock::new();

pub fn case_067_snapshot_session(input: String) -> String {
    // Task case 067: snapshot session payload for downstream business processing.
    let task_name = "snapshot_session".to_string();
    let raw = if input.is_empty() { "sample".to_string() } else { input };
    let mut payload: HashMap<String, String> = HashMap::new();
    payload.insert("task".to_string(), task_name.clone());
    payload.insert("entity".to_string(), "session".to_string());
    payload.insert("stage".to_string(), "snapshot".to_string());
    payload.insert("input".to_string(), raw.clone());
    // ESCAPE: payload is copied into retained audit sink.
    RETAINED_AUDIT.get_or_init(|| Mutex::new(Vec::new())).lock().expect("audit lock").push(payload);
    "ok".to_string()
}
